### ISSUE TITLE
Select broad bars in Barchart

### ISSUE THREAD

**Issue by Shawn3311** [grafana/grafana#47294]
**What happened**:
I tried to create a TimesSeries with a diagram containing a BarChart with "normal" broad bars.
When I did all the setting I only saw very thin bars which are only about 1 pixel broad.
I knew from pictures from the internet that broader bars are possible so I tried to find the correct setting.
After a long time of searching I compared the Panel-JSON of two different charts and found that the difference is only the label "spanNulls": false.
So I tried to fix this in the normal settings, but this is not possible when "style = bars" is chosen, because the feature  "Connect null values" is not visible in this mode.
![Grafana_01](https://user-images.githubusercontent.com/49248306/161633215-d666e179-fe54-40c0-abf6-48a0585ca528.png)

**What you expected to happen**:
I expected that I can select the feature "Connect null values = Always" also when "Style = Bars" is chosen. 
For me it's not logically that I have to switcht to "Style = Lines" to set the feature "Connect null values = Always" and the set Style back to "Bars" again. 
![Grafana_02](https://user-images.githubusercontent.com/49248306/161633298-925d4a25-064c-443b-89b7-0634db59a71f.png)
![Grafana_03](https://user-images.githubusercontent.com/49248306/161633310-751dbe44-a7eb-4331-8b5f-fc90152b888f.png)

**How to reproduce it (as minimally and precisely as possible)**:
Create a new chart with Style = Bars.

**Anything else we need to know?**:
none

**Environment**:
- Grafana version: 8.4.^1
- Data source type & version: InfluxDB 2.x
- OS Grafana is installed on: Raspbian Bullseye
- User OS & Browser: Windows 10 with Firefox
- Grafana plugins: none
- Others: none

**Comment by mellieA** [comment]
@Shawn3311 Are you still seeing this with 8.4.3+? I see a regression reported in https://github.com/grafana/grafana/issues/44513 that is fixed in 8.4.3. I tried this out on 8.4.5 and I'm not seeing the issue

**Comment by Shawn3311** [comment]
Hello,

 

I updated my system to Enterprise 8.4.5 and still see the issue.

I still have no access to “Connect null values” when Style=Bars is selected. Only when I switch to Style=Lines the Feature is visible.

 

Regards.

 

 

Von: Melori Arellano ***@***.***> 
Gesendet: Dienstag, 5. April 2022 20:03
An: grafana/grafana ***@***.***>
Cc: Shawn3311 ***@***.***>; Mention ***@***.***>
Betreff: Re: [grafana/grafana] Select broad bars in Barchart (Issue #47294)

 

@Shawn3311 <https://github.com/Shawn3311>  Are you still seeing this with 8.4.3+? I see a regression reported in #44513 <https://github.com/grafana/grafana/issues/44513>  that is fixed in 8.4.3. I tried this out on 8.4.5 and I'm not seeing the issue

**Comment by leeoniya** [comment]
this is reproducible with this dashboard:

<details><summary>skinny-bars.json</summary>

```json
{
  "annotations": {
    "list": [
      {
        "builtIn": 1,
        "datasource": {
          "type": "grafana",
          "uid": "-- Grafana --"
        },
        "enable": true,
        "hide": true,
        "iconColor": "rgba(0, 211, 255, 1)",
        "name": "Annotations & Alerts",
        "target": {
          "limit": 100,
          "matchAny": false,
          "tags": [],
          "type": "dashboard"
        },
        "type": "dashboard"
      }
    ]
  },
  "editable": true,
  "fiscalYearStartMonth": 0,
  "graphTooltip": 0,
  "id": 142,
  "links": [],
  "liveNow": false,
  "panels": [
    {
      "datasource": {
        "type": "testdata",
        "uid": "PD8C576611E62080A"
      },
      "fieldConfig": {
        "defaults": {
          "color": {
            "mode": "palette-classic"
          },
          "custom": {
            "axisLabel": "",
            "axisPlacement": "auto",
            "barAlignment": 0,
            "drawStyle": "line",
            "fillOpacity": 0,
            "gradientMode": "none",
            "hideFrom": {
              "legend": false,
              "tooltip": false,
              "viz": false
            },
            "lineInterpolation": "linear",
            "lineWidth": 1,
            "pointSize": 5,
            "scaleDistribution": {
              "type": "linear"
            },
            "showPoints": "auto",
            "spanNulls": false,
            "stacking": {
              "group": "A",
              "mode": "none"
            },
            "thresholdsStyle": {
              "mode": "off"
            }
          },
          "mappings": [],
          "thresholds": {
            "mode": "absolute",
            "steps": [
              {
                "color": "green",
                "value": null
              },
              {
                "color": "red",
                "value": 80
              }
            ]
          }
        },
        "overrides": [
          {
            "matcher": {
              "id": "byName",
              "options": "bars"
            },
            "properties": [
              {
                "id": "custom.drawStyle",
                "value": "bars"
              }
            ]
          }
        ]
      },
      "gridPos": {
        "h": 18,
        "w": 24,
        "x": 0,
        "y": 0
      },
      "id": 2,
      "options": {
        "legend": {
          "calcs": [],
          "displayMode": "list",
          "placement": "bottom"
        },
        "tooltip": {
          "mode": "single",
          "sort": "none"
        }
      },
      "targets": [
        {
          "alias": "bars",
          "datasource": {
            "type": "testdata",
            "uid": "PD8C576611E62080A"
          },
          "refId": "A",
          "scenarioId": "csv_metric_values",
          "stringInput": "70,null,70,null,70"
        },
        {
          "alias": "line",
          "datasource": {
            "type": "testdata",
            "uid": "PD8C576611E62080A"
          },
          "refId": "B",
          "scenarioId": "csv_metric_values",
          "stringInput": "1,20,90,30,5,0,1,20,90,30,5,0"
        }
      ],
      "title": "Panel Title",
      "type": "timeseries"
    }
  ],
  "schemaVersion": 36,
  "style": "dark",
  "tags": [],
  "templating": {
    "list": []
  },
  "time": {
    "from": "now-6h",
    "to": "now"
  },
  "timepicker": {},
  "timezone": "",
  "title": "skinny-bars",
  "uid": "6-QUBg87z",
  "version": 1,
  "weekStart": ""
}
```
</details>

have to see why the bar series is getting filled with `null` instead of `undefined` :thinking: 

![image](https://user-images.githubusercontent.com/43234/161859855-33dcd878-1f81-419d-8487-f5575b987ce0.png)

**Comment by Shawn3311** [comment]
Hello again,

 

sorry, but I don’t think the problem is solved. I’ve updated my system to 8.4.5 an now I always get skinny bars, when I set the time range to more than two days (one bar per day).

My goal is to have a bar chart for the last 30 days with broad bar and not only lines.

 

Thank you and regards.

 

Von: Leon Sorokin ***@***.***> 
Gesendet: Mittwoch, 6. April 2022 03:24
An: grafana/grafana ***@***.***>
Cc: Shawn3311 ***@***.***>; Mention ***@***.***>
Betreff: Re: [grafana/grafana] Select broad bars in Barchart (Issue #47294)

 

Closed #47294 <https://github.com/grafana/grafana/issues/47294>  via #47367 <https://github.com/grafana/grafana/pull/47367> .

**Comment by leeoniya** [comment]
> I’ve updated my system to 8.4.5 an now I always get skinny bars, when I set the time range to more than two days (one bar per day).

8.4.5 was released 7 days ago: https://github.com/grafana/grafana/releases/tag/v8.4.5

the fix was merged only 2 days ago. i thought i had it tagged properly for it make it into `v8.5.0-beta1` which was released today, but apparently that did not work as expected and i dont see the commit in [1]. so this fix will hopefully be in `v8.5.0-beta2`.

[1] https://github.com/grafana/grafana/compare/v8.5.0-beta1...v8.4.5

**Comment by leeoniya** [comment]
ok, looks like it was tagged properly after all, and should be part of https://grafana.com/grafana/download/8.5.0-beta1?edition=oss

Release branch includes that PR: https://github.com/grafana/grafana/commits/release-8.5.0-beta1
